File: CarTransportDashboard.web/src/app/models/Address.ts
export interface Address {
  companyName?: string;         // Optional business name
  addressLine1: string;         // Required: street address, building, etc.
  addressLine2?: string;        // Optional further detail
  locality: string;             // Town or city
  postalCode: string;
  country: string;

  lat?: number;
  lng?: number;
  formatted?: string;           // Full formatted address (e.g., from Google)
}

File: CarTransportDashboard/Models/Address.cs
namespace CarTransportDashboard.Models
{
    public class Address
    {
        public Guid Id { get; set; }
        public string? CompanyName { get; set; }
        public string AddressLine1 { get; set; } = string.Empty;
        public string? AddressLine2 { get; set; }
        public string Locality { get; set; } = string.Empty;
        public string PostalCode { get; set; } = string.Empty;
        public string Country { get; set; } = string.Empty;

        public double? Lat { get; set; }
        public double? Lng { get; set; }
        public string Formatted =>
           string.Join(", ", new[]
           {
                CompanyName,
                AddressLine1,
                AddressLine2,
                Locality,
                PostalCode,
                Country
           }.Where(p => !string.IsNullOrWhiteSpace(p)));


    }
}
